lucene/analysis/morfologik/src/java/org/apache/lucene/analysis/uk/UkrainianMorfologikAnalyzer.java

```diff
@@ -107,11 +107,18 @@ public UkrainianMorfologikAnalyzer(CharArraySet stopwords, CharArraySet stemExcl
   @Override
   protected Reader initReader(String fieldName, Reader reader) {
     NormalizeCharMap.Builder builder = new NormalizeCharMap.Builder();
+    // different apostrophes
     builder.add("\u2019", "'");
+    builder.add("\u0218", "'");
     builder.add("\u02BC", "'");
+    builder.add("`", "'");
+    builder.add("´", "'");
+    // ignored characters
     builder.add("\u0301", "");
-    NormalizeCharMap normMap = builder.build();
+    builder.add("\u00AD", "");
+    builder.add("\uFEFF", "");
```

Comment:
byte order mark shouldn't be replaced to nothing... if you have a byte order mark in your character input (reader) then your conversion from bytes is screwed up somewhere before.

Reply:
Just stumbled across this dicussion today. I'm the Wikimedia guy that arysin linked to above. And while it is _waaaaay_ to late, I just wanted to point out that while U+FEFF is used as the byte order mark at the beginning of a text stream, it is [also used](https://en.wikipedia.org/wiki/Byte_order_mark#Usage) as a "zero-width non-breaking space" within a text. That use was deprecated as of Unicode 3.2, but it is still quite common in practice on lots of Wikipedias. ICU normalization converts it to the empty string, which is how it usually gets handled on the wikis where we have had the chance to customize the analyzers, though that requires being able to open them up, so to speak.